File: Python3/1624. Largest Substring Between Two Equal Characters.py
class Solution:
    def maxLengthBetweenEqualCharacters(self, s: str) -> int:
        letter = dict()
        res = -1
        for i, c in enumerate(s):
            if c not in letter:
                letter[c] = i
            else:
                res = max(res, i - letter[c] - 1)
        
        return res

File: C++/1624. Largest Substring Between Two Equal Characters.cpp
class Solution {
public:
    int maxLengthBetweenEqualCharacters(string s) {
        int res = -1;
        map<char, int> letter_min;
        for(int i = 0; i < s.size(); ++i) {
            if(letter_min.count(s[i]) == 0) {
                letter_min[s[i]] = i;
            } else {
                res = max(res, i - letter_min[s[i]] - 1);
            }
        }
        return res;
    }
};
